PublishParasoftTestResults/publishtestresults.ts

```diff
@@ -64,6 +64,8 @@ if (isNullOrWhitespace(searchFolder)) {
 
 let xUnitReports: string[] = [];
 let sarifReports: string[] = [];
+let ruleIdsForWhoseCatIsGlobal: string[] = [];
```

Comment:
Use "new Set()" instead of string array. 
Although while looping through the <Rule> element, theoretically there should not be duplicates. But let's use set to be safe and as a habit for case like this.